confgenerator/config.go

```diff
@@ -149,6 +153,24 @@ func newValidator() *validator.Validate {
 		}
 		return t >= tmin
 	})
+	// time_denomination validates that the duration is whole to some time denomination
+	// eg. s, ms, m, h
+	v.RegisterValidation("whole_time_denomination", func(fl validator.FieldLevel) bool {
```

Comment:
The `panic` cases in this validator are not covered. Given that L148 (above in this file) is also not covered, do we accept the -0.7% fall in coverage due to these lines? We could think about adding tests that cover these lines, but not sure if that should live in this PR anyway since it's a new style of tests.

Reply:
I wouldn't worry about missing coverage for panics, they'll fail the tests anyway if they get triggered which is fine. IMO panic==assert and it should never be possible to write test cases to trigger them in the first place. That seems to be the case here so all seems good.

Reply:
+1